tests: drivers: uart: Aligned buffers to 32 bytes

sswetha18 opened the pull request:
This commit aligns the buffers to 32 bytes in
uart_async_api testcases.This is because most
of the platform DMA operations are aligned to
32 bytes.This ensures proper memory alignment
for cache handling and avoid potential
unaligned access issues.

sswetha18 committed:
tests: drivers: uart: Aligned buffers to 32 bytes

This commit aligns the buffers to 32 bytes in
uart_async_api testcases.This is because most
of the platform DMA operations are aligned to
32 bytes.This ensures proper memory alignment
for cache handling and avoid potential
unaligned access issues.

Signed-off-by: S Swetha <s.swetha@intel.com>

sswetha18 commented:
@dcpleung Could you please review this PR.

dcpleung commented on tests/drivers/uart/uart_async_api/boards/intel_rpl_s_crb.conf:
```diff
@@ -1 +1,2 @@
 CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_TEST_USERSPACE=n
```
Please separate this out into another commit with commit message stating why this is needed.

dcpleung replied:
You will also need to amend the commit message .

sswetha18 replied:
@dcpleung Changes done. please review

dcpleung commented on tests/drivers/uart/uart_async_api/src/test_uart_async.c:
```diff
@@ -104,8 +104,10 @@ static void uart_async_test_init(int idx)
 
 struct test_data {
 	volatile uint32_t tx_aborted_count;
+
 	__aligned(32) uint8_t rx_first_buffer[10];
 	uint32_t recv_bytes_first_buffer;
+
```
Please remove the addition of these two blank lines.

sswetha18 committed:
tests: drivers: uart: boards: Disable Userspace

This commit disables CONFIG_TEST_USERSPACE in
intel_rpl_s_crb.conf of uart_async_api
testcase.Enabling this config introduces
restrictions that interfere with cacheable
regions by blocking access and modifying
cache attributes.By disabling this
configuration, the following issues
are resolved:
-Cacheable region retain their attributes.
-Execution and data transactions work without
restrictions.
-System behavior align with expected
configuration in privileged mode.
-Some code primarily relying on non-cache
region continues to work.

This change is neccessary to ensure
cachebale memory regions function as intended
without interferance from user mode restrictions.

Signed-off-by: S Swetha <s.swetha@intel.com>

sswetha18 commented:
@dcpleung Changes are done. Please review

dcpleung commented on tests/drivers/uart/uart_async_api/prj.conf:
```diff
@@ -2,4 +2,4 @@ CONFIG_SERIAL=y
 CONFIG_UART_USE_RUNTIME_CONFIGURE=y
 CONFIG_ZTEST=y
-CONFIG_TEST_USERSPACE=y
+CONFIG_TEST_USERSPACE=n
```
No. This is not correct. You should not disable usespace testing for the whole test on every platforms. Do it in board overlay just for the boards needed.

sswetha18 commented:
@dcpleung Changes are done. Please review

sswetha18 commented:
@dcpleung Changes are done. Please review

sswetha18 commented:
Hi @dcpleung Changes are done. Could you please review the PR

sswetha18 commented:
@carlescufi Could you please review this PR

sswetha18 commented:
@jakub-uC @kartben Could you please review this PR
